frontend/src/components/RoutingMachine.tsx

```diff
@@ -0,0 +1,25 @@
+import L from "leaflet";
+import {createControlComponent} from "@react-leaflet/core";
+import "leaflet-routing-machine";
+import Location from "../types/Location";
+import "./RoutingMachine.css";
+
+export default function Routing({locations}: {
+    locations: Location[]
+}) {
+    const waypoints = locations.map(l => new L.LatLng(l.latitude, l.longitude));
+
+    const RoutingMachine = createControlComponent(() => L.Routing.control({
+        waypoints: waypoints,
+        lineOptions: {
+            styles: [{color: "#6FA1EC", weight: 4}],
+            extendToWaypoints: false,
+            missingRouteTolerance: 0
```

Comment:
I haven't taken a look at the RoutingMachine, so it's not like I know what missingRouteTolerance does, but setting tolerances to 0 is usually a bad idea. So my question is simply: have you checked and perhaps double-checked that this is fine?